src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -2992,6 +2993,30 @@ SubqueryPushdownMultiPlanTree(Query *queryTree)
 			(Node *) make_ands_implicit((Expr *) extendedOpNode->havingQual);
 	}
 
+	/*
+	 * Postgres standard planner evaluates expressions in the LIMIT/OFFSET clauses.
+	 * Since we're using original query here, we should manually evaluate the
+	 * expression on the LIMIT and OFFSET clauses. Note that logical optimizer
+	 * expects those clauses to be already evaluated.
+	 */
+	extendedOpNode->limitCount =
+		PartiallyEvaluateExpression(extendedOpNode->limitCount, NULL);
+	extendedOpNode->limitOffset =
+		PartiallyEvaluateExpression(extendedOpNode->limitOffset, NULL);
+
+	/*
+	 * If either LIMIT or OFFSET contains a parameter, PartiallyEvaluateExpression()
+	 * fails to evaluate them and we should error out sanely.
+	 */
+	if ((extendedOpNode->limitCount && !IsA(extendedOpNode->limitCount, Const)) ||
+		(extendedOpNode->limitOffset && !IsA(extendedOpNode->limitOffset, Const)))
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```

Comment:
While on that, we should have separate limit clause and limit offset errors, and error message could be re-written like
"unsupported limit expression" or "unsupported offset expression". We do not need to write a whole big sentence.